### ISSUE TITLE
nested interfaces do not get wired correctly and cause a `javac` compile error

### ISSUE THREAD

**Issue by mechite** [avaje/avaje-inject#922]
From testing in Discord...
Compilation fails due to package-private accessibility

...given an example from our use-case like such (note the package-private):
```java
@Dao // Datastax Cassandra Driver - a Data Access Object
interface SomeDao {


    // some code here
    @Select
    User findById(String id);


    // this Mapper from Datastax is for creating instances of SomeDao
    @Mapper
    interface SomeMapper {
        
        @DaoFactory
        SomeDao someDao(@DaoKeyspace String keyspace);
    }



    // and then... this Factory from Avaje... is for
    //                   creating instances of SomeMapper....
    //                   which makes SomeDao instances...
    @Factory 
    final class SomeMapperFactory {
        
        @Bean
        SomeMapper mapper(CqlSession session) {

            // DataStax Cassandra Driver - it is generating a builder for the @Mapper

            return new SomeDao_SomeMapperBuilder(session).build();
        }
    }
}
```

**Comment by SentryMan** [comment]
This turns out to be significantly harder to solve for than I thought, I'm thinking the move might be to just make a compile error saying factories can't be nested

**Comment by SentryMan** [comment]
so It's like technically possible, but I'd have to rewrite all sorts of things. also I'm seeing some freaky stuff like it somehow being processed three times. 

Non factory nested singletons work, it's specifically factories that have this issue.
